src/server/routes/contract/read/read-batch.ts

```diff
@@ -0,0 +1,170 @@
+  fastify.route<RouteGeneric>({
+    method: "POST",
+    url: "/contract/:chain/read-batch",
+    schema: {
+      summary: "Batch read from multiple contracts",
+      description:
+        "Execute multiple contract read operations in a single call using Multicall3",
+      tags: ["Contract"],
+      operationId: "readMulticall",
+      params: paramsSchema,
+      body: readMulticallRequestSchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseSchema,
+      },
+    },
+    handler: async (request, reply) => {
+      const { chain: chainSlug } = request.params;
+      const { calls, multicallAddress = MULTICALL3_ADDRESS } = request.body;
+
+      const chainId = await getChainIdFromChain(chainSlug);
+      const chain = await getChain(chainId);
+
+      try {
+        // Encode each read call
+        const encodedCalls = await Promise.all(
+          calls.map(async (call) => {
+            const contract = await getContract({
+              client: thirdwebClient,
+              chain,
+              address: call.contractAddress,
+            });
+
+            const method =
+              (call.functionAbi as unknown as AbiFunction) ??
+              (await resolveMethod(call.functionName)(contract));
+
+            const transaction = prepareContractCall({
+              contract,
+              method,
+              params: call.args || [],
+              // stubbing gas values so that the call can be encoded
+              maxFeePerGas: 30n,
+              maxPriorityFeePerGas: 1n,
+              value: 0n,
+            });
+
+            const calldata = await resolvePromisedValue(transaction.data);
+            if (!calldata) {
+              throw new Error("Failed to encode call data");
+            }
+
+            return {
+              target: call.contractAddress,
+              abiFunction: method,
+              allowFailure: true,
+              callData: calldata,
+            };
+          }),
+        );
+
+        // Get Multicall3 contract
+        const multicall = await getContract({
+          chain,
+          address: multicallAddress,
+          client: thirdwebClient,
+        });
+
+        // Execute batch read
+        const results = await readContract({
+          contract: multicall,
+          method: MULTICALL3_AGGREGATE_ABI,
+          params: [encodedCalls],
+        });
+
+        // Process results
+        const processedResults = results.map((result: unknown, i) => {
+          const { success, returnData } = result as {
+            success: boolean;
+            returnData: unknown;
+          };
+
+          const [_sig, _inputs, outputs] = prepareMethod(
+            encodedCalls[i].abiFunction,
+          );
+
+          const decoded = decodeAbiParameters(
+            outputs,
+            returnData as `0x${string}`,
+          );
+
+          return {
+            success,
+            result: success ? bigNumberReplacer(decoded) : null,
```

Comment:
We should not need bigNumberReplacer. We're on v5 SDK which does not rely on ethers.